### PR TITLE
Add support for Smith fuzzer

### PR DESCRIPTION
**Pull request by ruchirK**
This change uses the smith fuzzing api https://jibson.dev/smith/ and is meant to be a starting point towards a fuzzing infra that we can run nightly

This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/2754)

### CODE REVIEW

File: demo/smith_fuzz/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+[package]
+name = "smith-fuzz"
+description = "Fuzzer intergration using Smith to generate Materialize inputs"
+version = "0.1.0"
+authors = ["Brandon W Maister <bwm@materialize.io>"]
+edition = "2018"
+publish = false
+
+[dependencies]
```
Comment (ruchirK) [inline demo/smith_fuzz/Cargo.toml:9]: audit these dependencies for the ones we actually need

File: demo/smith_fuzz/ci/mzbuild.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+name: smith-fuzz
```
Comment (ruchirK) [inline demo/smith_fuzz/ci/mzbuild.yml:10]: I'm not familiar enough with our current ci infra so not sure what exactly gets built as a result of these files -- make sure its what I want

**Comment by benesch** [comment]
What still needs to get sorted for this to run more on a regular schedule? Buildkite makes scheduling nightly jobs super easy. 

As a general rule, I like to keep non-determinism out of PR CI builds, so I don't think we should be hitting jibson.dev on every CI build. (If you can mock it out, though, then go for it.)

**Comment by ruchirK** [comment]
> What still needs to get sorted for this to run more on a regular schedule? Buildkite makes scheduling nightly jobs super easy.

2 bits -- I don't know how to set it up :) and I'm not sure what current constraints are on request load. I'm talking to matt later today so should get answers for the 2nd bit

in the meantime would you mind if I merged this as a first step?

**Pull request merged**